### ISSUE TITLE
Cannot enable 'outgoing' multilink without 'incoming' multilink

### ISSUE THREAD

**Issue by fuzzypixelz** [eclipse-zenoh/zenoh#1533]
### Describe the bug

In order for a peer to establish multiple links to a router, it needs to set `transport/unicast/multilink` to a value `> 1`. This means that said peer would also be accepting multiple links from other peers.

Thus there is no way to enable multilink for 'outgoing' connections without inadvertently doing so for 'incoming' connections.

### To reproduce

N/A

### System info

- ref: ae9719f

**Comment by Mallets** [comment]
As far as I recall, no limitations are in place for outgoing links. So a peer can always open as many links it wants. The limit is on the incoming links.

**Comment by sashacmc** [comment]
We decided not to change it, but to document it correctly.
@milyin could you please take care about it?

**Comment by milyin** [comment]
After discussion with @fuzzypixelz :

This test https://github.com/eclipse-zenoh/zenoh/blob/4cfe94aae4c75d35e1f6b149cb08d9fb7be5bfe6/io/zenoh-transport/src/unicast/manager.rs#L263 uses `max_links > 1` condition to disable/enable multiple *outgoing* links. 
So this is actual issue: the same parameter serves both as 
- maximum incoming links value
- boolean enable parameter for multiple outgoing links

This makes sense to separate this

**Comment by fuzzypixelz** [comment]
@sashacmc I think we should move to update the configuration format and not only update the documentation.

**Comment by oteffahi** [comment]
IMHO `max_links` should serve to limit both. The distinction between incoming and outgoing doesn't make sense to me, we're interested in the total number of links established between two instances.
A config which limits one direction or the other will be hard to put in place, since establishment between peers/routers can go in both ways and the only configuration we offer currently to decide which should connect to the other is based on the ZID (comparing two random identifiers selected independently by each side on startup).

**Comment by sashacmc** [comment]
> [@sashacmc](https://github.com/sashacmc) I think we should move to update the configuration format and not only update the documentation.

Yes, we have big configuration rework plans for the next year, but for now we should at least properly document it.
